Include/PGE/String/String.h

```diff
@@ -287,47 +290,68 @@ class String {
         bool isEmpty() const;
 
     private:
-        String(int size);
+        String(nullptr_t) { } // Dummy constructor
         String(const char* cstr, size_t size);
         String(const String& other, int from, int cnt);
 
-        static void copy(String& dst, const String& src);
-
-        static constexpr int SHORT_STR_CAPACITY = 16;
+        static constexpr int SHORT_STR_CAPACITY = 40;
 
         struct Data {
-            int strByteLength = -1;
-
-            int cCapacity = SHORT_STR_CAPACITY;
-
             // Lazily evaluated.
-            mutable u64 _hashCode;
+            mutable u64 _hashCode = 0;
             mutable int _strLength = -1;
-            mutable bool _hashCodeEvaluted = false;
+
+            int strByteLength = -1;
+        };
+
+        struct CoreInfo {
+            char* chs;
```

Comment:
Rename this

Reply:
What to?

Reply:
`characterBuffer`, `charBuf`, `cstrPtr`, anything that says more than `chs`